src/adapter/src/catalog/open.rs

```diff
@@ -771,10 +771,14 @@ impl Catalog {
                 ];
                 let to_migrate: Vec<_> = txn
                     .get_system_items()
+                    // Filter to the relevant sources.
                     .filter(|mapping| {
                         mapping.description.schema_name == MZ_INTERNAL_SCHEMA && candidates.contains(&mapping.description.object_name.as_str())
                     })
+                    // Map down to the GlobalId.
                     .map(|mapping| mapping.unique_identifier.id)
+                    // Filter to only items that exist and don't already exist in the migrated set.
+                    .filter(|id| state.try_get_entry(id).is_some() && !migrated_builtins.contains(id))
```

Comment:
How do these tables not exist?

Reply:
Concur we should understand what's going on before merging.

Reply:
@jkosh44 and I looked into this by re-running the upgrade check and dumping the catalog of an effected environment. What's happening is one of these sources used to exist as a `Table` and has since been changed to a `Source`. The builtin migration saw the `Source` as a new object and thus never issued a retraction for the previous `Table`, so the old ID is hanging around in the system objects collection. Later then we create the in-memory state for the catalog we ignore the old table since it doesn't exist as an item in `builtin.rs`.